lib/systems/lobby_system.cpp

```diff
@@ -0,0 +1,148 @@
+                    // Check if character is available
+                    bool available = true;
+                    for(auto character : picked_characters) {
+                        if(character.first == selector->selected_character && character.second) {
+                            available = false;
+                            break;
+                        }
+                    }
+
+                    if(available) {
+                        if(selector->selected_character != Character::RANDOM) {
+                            for(auto& character : picked_characters) {
+                                if(character.first == selector->selected_character) {
+                                    character.second = true;
+                                    break;
+                                }
+                            }
+
+                            // Update the other selector so they do not show the character chosen by this player
+                            auto selector_entities_2 = em->getEntitiesByComponent<CharacterSelectionComponent>();
+                            for(auto& [entity_id_2, selector_2] : selector_entities_2) {
+                                if(entity_id != entity_id_2 && selector_2->joined && !selector_2->picked && selector_2->selected_character == selector->selected_character) {
+                                    // This is not the current selector, the player has joined but not picked a character and the character currently displayed is just picked by the other player
+                                    for(auto& character : picked_characters) {
+                                    // Select the first available character
+                                    if(!character.second) {
+                                        selector_2->selected_character = character.first;
+                                        ef->changeCharacterSelectorTexture(entity_id_2, character.first, true);
+                                        break;
+                                    }
+                                }
+                                }
+                            }
+                        }
+                        selector->picked = true;
+
+                        auto transform_component = em->getComponent<TransformComponent>(entity_id);
+                        selector->player_entity_id = ef->createPlayer(selector->player_id, selector->selected_character, transform_component->x_pos, transform_component->y_pos);
+
+                        ef->createImage("menu/check.png", transform_component->x_pos, transform_component->y_pos, 84, 66, Layers::Foreground, 255);
+
+                        input.removeTimeToWait(selector->player_id, PlayerInput::X_AXIS);
+                    }
+                } else {
+                    // The player already picked a character
+                    continue;
+                }
+            }
+        }
+
+        bool changed = false;
```

Comment:
I would put a comment here to say that this is the code that covers selecting a character.